src/dir.rs

```diff
@@ -468,10 +468,76 @@ impl Drop for TempDir {
     }
 }
 
-pub(crate) fn create(path: PathBuf) -> io::Result<TempDir> {
-    fs::create_dir(&path)
-        .with_err_path(|| &path)
-        .map(|_| TempDir {
-            path: path.into_boxed_path(),
-        })
+pub(crate) fn create(
+    path: PathBuf,
+    permissions: Option<&std::fs::Permissions>,
+) -> io::Result<TempDir> {
+    imp::create(path, permissions)
+}
+
+mod imp {
```

Comment:
I'd prefer it if we created the entire directory structure.

Reply:
I can do that, even though it's a change that might have performance implications. For that reason alone I wouldn't do it without also introducing a new method for it, and do so in a separate PR.

Reply:
Wait, no. I mean: put the modules in different files/directories (`imp/...`). Sorry, I realize that that was confusing.

Reply:
Ah, right, I could have guessed it. And it's done now. A note regarding naming: It feels like `any` should be `other`, but I decided not to do that as `other` means unsupported in the `file` folder, while here it is a catch-all.

Reply:
Hm. Yeah, I may go through and do a rename later but those are implementation details so.. meh.